test/functional/wallet_send.py

```diff
@@ -224,8 +226,10 @@ def run_test(self):
         assert_equal(self.nodes[1].decodepsbt(res1["psbt"])["fee"],
                      self.nodes[1].decodepsbt(res2["psbt"])["fee"])
         # but not at the same time
-        self.test_send(from_wallet=w0, to_wallet=w1, amount=1, arg_conf_target=1, arg_estimate_mode="economical",
-                       conf_target=1, estimate_mode="economical", add_to_wallet=False, expect_error=(-8,"Use either conf_target and estimate_mode or the options dictionary to control fee rate"))
+        for mode in ["unset", "economical", "conservative", "btc/kb", "sat/b"]:
+            self.test_send(from_wallet=w0, to_wallet=w1, amount=1, arg_conf_target=1, arg_estimate_mode="economical",
```

Comment:
I noticed that both `arg_estimate_mode` and `estimate_mode` appear in these test parameters. That seems odd. I wonder whether both pertain to the same value. If one of them would supersede the other, and that happened to be `arg_estimate_mode`, we would actually only be testing for `economical` here. If they actually do refer both to the same value, they should probably both be set to `=mode`.

Reply:
Good point. In RPC send, `conf_target` and `estimate_mode` can be both args (`arg_estimate_mode` in this test file, which sets up its own mini-framework) and options (`estimate_mode` in the test).

There seem to be some oddities or bugs caused by this, e.g. the commented-out tests in lines 289-292. The send RPC is marked as experimental, but these should be fixed or made consistent with the other RPCs after this is merged. I started to look into it but didn't find a fix quickly, and the PR is already quite large and the deadline for 0.21 branch-off this weekend is looming.

Reply:
Okay, since this is only in a test, it may be fine to keep track of it, but not get too hung up on fixing every little bit before the merge.